File: playground/backend/mapper.js
function toRow(profile) {
  const created = profile.created_at;
  const githubCreatedAt = created
    ? created.replace("T", " ").replace("Z", "")
    : null;
  return {
    github_id: profile.id,
    login: profile.login,
    name: profile.name ?? null,
    avatar_url: profile.avatar_url ?? null,
    bio: profile.bio ?? null,
    company: profile.company ?? null,
    location: profile.location ?? null,
    public_repos: profile.public_repos ?? 0,
    followers: profile.followers ?? 0,
    following: profile.following ?? 0,
    github_created_at: githubCreatedAt,
  };
}

module.exports = { toRow };

File: services/profile-service/mapper.go
package main

import "strings"

func ToRow(u GithubUser) Profile {
	created := ""
	if u.CreatedAt != "" {
		created = strings.Replace(strings.Replace(u.CreatedAt, "T", " ", 1), "Z", "", 1)
	}
	return Profile{
		GithubID: u.ID, Login: u.Login, Name: u.Name, AvatarURL: u.AvatarURL,
		Bio: u.Bio, Company: u.Company, Location: u.Location,
		PublicRepos: u.PublicRepos, Followers: u.Followers, Following: u.Following,
		GithubCreatedAt: created,
	}
}
